### PR TITLE
[machine] virtual app set state integration test

### PR DESCRIPTION
**Pull request by IIIIllllIIIIllllIIIIllllIIIIllllIIIIll**
This PR adds an integration test for `virtual-app-set-state-commitment`

### CODE REVIEW

File: packages/contracts/test/app-registry.spec.ts [hunk 1]
```diff
@@ -50,7 +50,9 @@ describe("AppRegistry", () => {
     provider = waffle.createMockProvider();
     wallet = (await waffle.getWallets(provider))[0];
 
-    appRegistry = await waffle.deployContract(wallet, AppRegistry);
+    appRegistry = await waffle.deployContract(wallet, AppRegistry, [], {
+      gasLimit: 6000000
```
Comment (snario) [inline packages/contracts/test/app-registry.spec.ts]: Why is this needed?

File: packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+let appRegistry: Contract;
+
+let signingKeys: SigningKey[];
+let targetAppInstance: AppInstance;
+let c0: VirtualAppSetStateCommitment;
+let s0: Signature;
+
+expect.extend({ toBeEq });
+
+const rejectionOf = (promise: Promise<any>) =>
```
Comment (snario) [inline packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts]: `ethereum-waffle` offers a matcher: `await expect(<transaction).to.revertWith(...)`

File: packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    );
+  });
+  it("fails with nonce above expiry", async () => {
+    const c1 = new VirtualAppSetStateCommitment(
+      network,
+      targetAppInstance.identity,
+      65536,
+      targetAppInstance.timeout,
+      targetAppInstance.hashOfLatestState,
+      65537
```
Comment (snario) [inline packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts]: There are several magic numbers in here, can you make them more clear w.r.t. intent

File: packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    const contractAppState = await appRegistry.appStates(
+      targetAppInstance.identityHash
+    );
+
+    expect(contractAppState.appStateHash).toBe(
+      targetAppInstance.hashOfLatestState
+    );
+  });
+  it("fails with nonce above expiry", async () => {
+    const c1 = new VirtualAppSetStateCommitment(
```
Comment (snario) [inline packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts]: Better variable name than `c1`?

File: packages/contracts/test/app-registry.spec.ts (hunk 1 above)
Comment (IIIIllllIIIIllllIIIIllllIIIIllllIIIIll) [reply packages/contracts/test/app-registry.spec.ts]: the default gas limit is fixed at 4 million (it is not estimated) which is too low to deploy `AppRegistry`

Comment (snario) [reply packages/contracts/test/app-registry.spec.ts]: Worth noting this in a comment in the code imo

File: packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts (hunk 2 above)
Comment (snario) [reply packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts]: You can still import `solidity` from `ethereum-waffle` and use `chai.use(solidity)`

Comment (snario) [reply packages/machine/test/integration/virtual-app-set-state-commitment.spec.ts]: Waffle is not a testing framework. Waffle's helper functions work well when used within `mocha` and has issues with `jest` simply because of the asynchronicity of it I believe. It might work with `jest` with `runInBand` on true.

In any case, you don't need to care about this to use the matchers.
